raster-foundry/raster-foundry
# Add help center to show customer specific videos & docs

## Overview

This PR adds a new dropdown on the navigation bar that plays tutorial videos within RF and links to customer specific documentations. Both app-wide and page-specific links/resources are enabled dynamically based on state, which are defined in `help_docs.json` in the same directory as `overrides.js`.

~This PR is currently WIP due to: 1. need to talk to @designmatty about css best practices; 2. need to talk to @jmorrison1847 about docs and pages.~

**Updates on Mar. 26th:**
- Talked to @designmatty offline to 
    * fix some css issues
    * see if we should implement video dragging feature for this card, and decided not to -- a cool feature for future though.
- Made some changes to have existing features make more sense.
- Note for testing: be sure to use the updated template of `overrides.js` to load `help_docs.json`

**Updates on Mar. 27th:**
- Talked to @jmorrison184 and decided that links in `help_docs.json` for this card would only serve as examples for adding more in the future. -- Thus, removed the `WIP` tag.

**Known issues when playing youtube videos:**
* `Failed to execute 'postMessage' on 'DOMWindow'` in console: this is an issue with Youtube API as mentioned in [here](https://github.com/sampotts/plyr/issues/631)
* `Error parsing header X-XSS-Protection: 1;` in console: this is an issue in chrome browsers as mentioned in [here](https://github.com/CookPete/react-player/issues/288)
* You may see error messages like `content blocked by user` in console -- this might be that your browser has an ad blocker plugin blocking youtube ads.
* **All above mentioned issues should not affect video playing**

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo

#### Play video -> collapse -> expand -> close

![video-play-close](https://raw.githubusercontent.com/aaronxsu/MyData/master/gifs/rf-pr-gifs/video-play-close.gif)

#### Play video -> collapse -> close in (fake) mini mode (you can hear the sound though)

![mini-close](https://raw.githubusercontent.com/aaronxsu/MyData/master/gifs/rf-pr-gifs/mini-close.gif)

#### App-wide and page-specific help docs

![state-doc-link](https://raw.githubusercontent.com/aaronxsu/MyData/master/gifs/rf-pr-gifs/state-doc-link.gif)

## Note:

Will probably need a feature flag.


## Testing Instructions

 * Go to whatever pages with a navigation bar presented, check if it has app-wide links and videos in the help center, play with the video feature.
 * Navigate to different pages to see if page specific docs and videos are listed correctly in the help center.

Closes #3076
